rust/lance/src/dataset/scanner.rs

```diff
@@ -542,7 +545,13 @@ impl Scanner {
         let output_schema = self.output_schema()?;
         let remaining_schema = output_schema.exclude(plan.schema().as_ref())?;
         if !remaining_schema.fields.is_empty() {
+            if order_by_distance {
+                plan = self.order_by(ROW_ID, plan)?;
+            }
             plan = self.take(plan, &remaining_schema, self.batch_readahead)?;
+            if order_by_distance {
+                plan = self.order_by(DIST_COL, plan)?;
+            }
```

Comment:
Hmm I think it would be more optimal to push down this sorting into take so we only have to perform it on the new columns, like we do for the user-facing `take` implementation. As-is, this will sort the vector column, which is known to be an expensive CPU-bound operation, even on bathes of 1000 rows.

Reply:
That makes sense.  Could we just sort the row ids right here: https://github.com/lancedb/lance/blob/cba8ecbb41ba7f484c4aec29447f50135e040cc5/rust/lance/src/io/exec/take.rs#L122 ?

Reply:
Lemme try

Reply:
Took more time to think about this. depend on where you sort the data, it might involve either memcpy of vectors (if you do the sorting before take, or need  memcpy of other columns (i.e., images and etc) after `take_rows()`.